### PR TITLE
Add documentation for sending requests with an absolute URI

### PR DESCRIPTION
**Pull request by ikhoon**

### CODE REVIEW

**Comment by ikhoon** [comment]
image

File: site/src/pages/docs/client-http.mdx [hunk 1]
```diff
@@ -1,5 +1,7 @@
 # Calling an HTTP service
 
+You can create <type://WebClient> with a base URI and send a request with a relative path.
```
Comment (trustin) [inline site/src/pages/docs/client-http.mdx:3]: `a <type://WebClient>`

File: site/src/pages/docs/client-http.mdx [hunk 2]
```diff
@@ -23,6 +25,43 @@ AggregatedHttpResponse res3 = webClient.post("/upload", "{ \"foo\": \"bar\" }")
                                        .aggregate().join();
 ```
 
+You can also create <type://WebClient> without a base URI and send a request with an absolute URI.
+The ad-hoc <type://WebClient> would be useful in the following cases:
+
+- Send requests to arbitrary endpoints
+- Build a proxy server
+- Handle redirected requests
```
Comment (trustin) [inline site/src/pages/docs/client-http.mdx:33]: Sending/Building/Handling?

File: site/src/pages/docs/client-http.mdx [hunk 3]
```diff
@@ -23,6 +25,43 @@ AggregatedHttpResponse res3 = webClient.post("/upload", "{ \"foo\": \"bar\" }")
...
+import com.linecorp.armeria.common.HttpStatus;
+import com.linecorp.armeria.common.HttpHeaderNames;
+
+// Create a WebClient without a base URI.
+WebClient webClient = WebClient.of();
+
+// Send a request with an absolute URI.
+AggregatedHttpResponse res1 = webClient.get("http://example.com/foo/bar.txt")
+                                       .aggregate()
+                                       .join();
```
Comment (trustin) [inline site/src/pages/docs/client-http.mdx:45]: @tobias- told me this did not work for him. Could you double-check if this example works as expected?

Comment (ikhoon) [reply site/src/pages/docs/client-http.mdx:45]: Sure! Let me check it.

Comment (ikhoon) [reply site/src/pages/docs/client-http.mdx:45]: Tested and it worked as expected. 😀
Currently ad-hoc `WebClient` is verified by:
https://github.com/line/armeria/blob/fbe90f0f2a952578012a552c26b3f4f8bdb33c8c/core/src/test/java/com/linecorp/armeria/client/HttpClientRequestPathTest.java#L107-L118

**Review by minwoox — APPROVED** [review]
Thanks a lot, @ikhoon!

**Review by trustin — APPROVED** [review]

**Pull request merged**